target_chains/ton/contracts/contracts/Pyth.fc

```diff
@@ -1,27 +1,46 @@
 #include "imports/stdlib.fc";
-#include "Wormhole.fc";
+#include "common/errors.fc";
+#include "common/storage.fc";
 
 ;; Opcodes
 const int OP_UPDATE_GUARDIAN_SET = 1;
 const int OP_EXECUTE_GOVERNANCE_ACTION = 2;
 
-;; Internal message handler
-() recv_internal(int my_balance, int msg_value, cell in_msg_full, slice in_msg_body) impure {
-    if (in_msg_body.slice_empty?()) { ;; ignore empty messages
-        return ();
-    }
+(int, int, int, int) parse_price(slice price_feed) {
+    int price = price_feed~load_int(256);
+    int conf = price_feed~load_uint(64);
+    int expo = price_feed~load_int(32);
+    int publish_time = price_feed~load_uint(64);
+    return (price, conf, expo, publish_time);
+}
+
+(int, int, int, int) price_unsafe(int price_feed_id) method_id {
```

Comment:
for the future: if these are going to be exposed to the users, can you add proper docs to them?

Reply:
will do